src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,100 @@
+
+    @Test
+    @DisplayName("User register with null login")
+    void register_nullLogin_throwNullPointerException() {
+        User userWithNullLogin = new User(null, "1234567", 45);
+        assertThrows(NullPointerException.class,
+                     () -> registrationService.register(userWithNullLogin));
+    }
+
+    @Test
+    @DisplayName("User register with null password")
+    void register_nullPassword_throwNullPointerException() {
+        User userWithNullPassword = new User("ivanivanov12@gmail.com", null, 45);
+        assertThrows(NullPointerException.class,
+                     () -> registrationService.register(userWithNullPassword));
+    }
+
+    @Test
+    @DisplayName("User register with null age")
+    void register_nullAge_throwNullPointerException() {
+        User userWithNullAge = new User("ivanivanov12@gmail.com", "1234567", null);
+        assertThrows(NullPointerException.class,
+                     () -> registrationService.register(userWithNullAge));
+    }
+
+    @Test
+    @DisplayName("User register correct user case #1")
+    void register_correctUser_ok() {
+        User user = new User("mykhailo777@gmail.com", "1234567", 23);
+        User actualUser = registrationService.register(user);
+        User expectedUser = new User("mykhailo777@gmail.com", "1234567", 23);
+        assertEquals(actualUser, expectedUser);
+    }
+
+    @Test
+    @DisplayName("User register correct user case #2")
+    void register_correctUserSecondCase_ok() {
+        User user = new User("ivangolybev@gmail.com", "1234567", 28);
+        User actualUser = registrationService.register(user);
+        User expectedUser = new User("ivangolybev@gmail.com", "1234567", 28);
+        assertEquals(actualUser, expectedUser);
+    }
+
+    @Test
+    @DisplayName("User register correct user case #3")
+    void register_correctUserThirdCase_ok() {
+        User user = new User("oleksiiivanov@gmail.com", "1234567", 33);
+        User actualUser = registrationService.register(user);
+        User expectedUser = new User("oleksiiivanov@gmail.com", "1234567", 33);
+        assertEquals(actualUser, expectedUser);
+    }
+
+    @Test
+    @DisplayName("User login already exists register")
+    void register_userLoginExist_throwUserLoginExistException() {
+        User user = new User("mykhailo88525@gmail.com", "1234567", 23);
+        User actualUser = registrationService.register(user);
+        assertThrows(UserLoginExistsException.class,
+                     () -> registrationService.register(actualUser));
+    }
+
+    @Test
+    @DisplayName("User password length less than minimum")
+    void register_userPasswordLessThanMinimum_throwUserPasswordLengthException() {
+        User userWithIncorrectPassword = new User("mykhailo12@gmail.com", "123", 82);
+        assertThrows(UserPasswordLengthException.class,
+                     () -> registrationService.register(userWithIncorrectPassword));
```

Comment:
Make sure to test edge cases for the password length. You are testing for a password length of 3, but you should also test for lengths of 5 (just below the minimum) and 6 (exactly the minimum) to ensure your validation works correctly.